### PR TITLE
fix: response headers serialization

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**

### CODE REVIEW

**Review by Uzlopak — COMMENTED** [review]
Maybe the for const of is not nice?

File: lib/response.js [hunk 1]
```diff
@@ -155,16 +155,26 @@ function getNullSocket () {
 }
 
 function copyHeaders (response) {
-  response._lightMyRequest.headers = Object.assign({}, response.getHeaders())
+  const headers = Object.assign({}, response.getHeaders())
+
+  for (const [name, value] of Object.entries(headers)) {
```
[the remark this answers is not in the document]
Comment (mcollina) [reply lib/response.js:160]: I think it's pretty bad :(

File: lib/response.js [hunk 2]
```diff
@@ -155,16 +155,26 @@ function getNullSocket () {
...
+    if (Array.isArray(value)) {
+      headers[name] = value.map(value => '' + value)
+    } else {
+      headers[name] = '' + value
+    }
+  }
```
Comment (mcollina) [inline lib/response.js:166]: I don't think this should be done inside `copyHeaders`, but rather right before generating the payload: https://github.com/fastify/light-my-request/pull/248/files#diff-c03f99ad1a5cf68a9b32468371973a5898c4c9c9d5f097e124f1d2b7b27db11dR118.

The reason is that if multiple `res.setHeader()` are issued, they will not be picked up.

File: lib/response.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcollina) [reply lib/response.js:160]: It would be better to use `Object.keys()` instead.

**Comment by ivan-tymoshenko** [comment]
I pushed a fixup commit. Idk where it is.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**